OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -89,13 +89,17 @@ Order IIssueDeployOrder.IssueDeployOrder(Actor self, bool queued)
 
 		void IResolveOrder.ResolveOrder(Actor self, Order order)
 		{
+			if (order.OrderString != "BeginMinefield" && order.OrderString != "PlaceMinefield" && order.OrderString != "PlaceMine")
+				return;
+
+			var cell = self.World.Map.CellContaining(order.Target.CenterPosition);
 			if (order.OrderString == "BeginMinefield")
-				minefieldStart = order.TargetLocation;
+				minefieldStart = cell;
 
 			if (order.OrderString == "PlaceMine")
```

Comment:
Minor scope-creep: You could make this and the "PlaceMinefield" check below an `else if` while you're at it.